cl_manycore/libraries/bsg_manycore_cuda.cpp

```diff
@@ -47,312 +56,858 @@ static int hb_mc_get_y(tile_t *tiles, uint8_t *y_list, uint32_t num_tiles) {
 		return HB_MC_FAIL;
 	}
 	for (int i = 0; i < num_tiles; i++) {
-		y_list[i] = tiles[i].y;
+		y_list[i] = tiles[i].coord.y;
 	}
 	return HB_MC_SUCCESS;
 }
 
-/*!
- * creates a awsbwhal::MemoryManager object and stores it in a global table.
- * @param eva_id which specifies which EVA <-> NPA mapping.
- * @param elf path to ELF binary
- * @return HB_MC_SUCCESS on success and HB_MC_FAIL on failure.
+
+
+
+
+/**
+ * Takes in a device_t struct and initializes a mesh of tile in the Manycore device.
+ * @param[in]  device        Pointer to device
+ * @parma[in]  dim           X/Y dimensions of the tile pool (mesh) to be initialized
+ * @return HB_MC_SUCCESS on success and HB_MC_FAIL on failure. 
  */
-static int hb_mc_create_memory_manager (eva_id_t eva_id, char *elf) {
-	eva_t program_end;
-	if (symbol_to_eva(elf, "_bsg_dram_end_addr", &program_end) != HB_MC_SUCCESS) {
+int hb_mc_mesh_init (device_t *device, hb_mc_dimension_t dim){ 
+
+	int error;
+
+	const hb_mc_config_t *cfg = hb_mc_manycore_get_config(device->mc);
+	hb_mc_dimension_t device_dim = cfg->dimension;
+
+	
+	if (dim.x <= 0){
+		bsg_pr_err("%s: Mesh X dimension (%d) not valid.\n", __func__, dim.x); 
 		return HB_MC_FAIL;
 	}
-	uint32_t alignment = 32;
-	uint32_t start = program_end + alignment - (program_end % alignment); /* start at the next aligned block */
-	uint32_t size = DRAM_SIZE;
-	mem_manager[eva_id] = new awsbwhal::MemoryManager(DRAM_SIZE, start, alignment); 
+	if (dim.y <= 0){
+		bsg_pr_err("%s: Mesh Y dimension (%d) not valid.\n", __func__, dim.y);
+		return HB_MC_FAIL;
+	}
+	if (dim.x > device_dim.x){
+		bsg_pr_err("%s: Mesh X dimension (%d) larger than device X dimension.\n", __func__, dim.x, device_dim.x); 
+		return HB_MC_FAIL;
+	}
+	if (dim.y > device_dim.y){
+		bsg_pr_err("%s: Mesh Y dimension (%d) larger than device Y dimension.\n", __func__, dim.y, device_dim.y);
+		return HB_MC_FAIL;
+	}
+
+	tile_t* tiles = new tile_t [dim.x * dim.y];
+	for (int x = default_origin.x; x < default_origin.x + dim.x; x++){
+		for (int y = default_origin.y; y < default_origin.y + dim.y; y++){
+			int tile_id = (y - default_origin.y) * dim.x + (x - default_origin.x);
+			tiles[tile_id].coord.x = x;
+			tiles[tile_id].coord.y = y;
+			tiles[tile_id].origin.x = default_origin.x;
+			tiles[tile_id].origin.y = default_origin.y;
+			tiles[tile_id].tile_group_id.x = -1;
+			tiles[tile_id].tile_group_id.y = -1;
+			tiles[tile_id].free = 1;
+		}
+	}
+
+	mesh_t *mesh = new mesh_t; 
+	mesh->dim.x = dim.x;
```

Comment:
Please use the setters and getters here too please.

Reply:
Thanks for spotting it! Fixed.